util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java

```diff
@@ -186,12 +216,14 @@ static URI parseProxyAddress(@Nullable String proxyAddress) {
 
   /** Wrapper class to have socketKeepAlive property while creating the socket */
   @VisibleForTesting
-  static class SslKeepAliveSocketFactory extends SSLSocketFactory {
+  static final class CustomSslSocketFactory extends SSLSocketFactory {
 
     private final SSLSocketFactory wrappedSockedFactory;
+    private final Integer readTimeoutMillis;
 
-    public SslKeepAliveSocketFactory(SSLSocketFactory wrappedSocketFactory) {
+    public CustomSslSocketFactory(SSLSocketFactory wrappedSocketFactory, Duration readTimeout) {
       this.wrappedSockedFactory = wrappedSocketFactory;
+      this.readTimeoutMillis = readTimeout != null ? (int) readTimeout.toMillis() : null;
```

Comment:
Probably will be safer to use `Math.toExactInt` instead of `(int)` cast

Reply:
Done, thanks!